manila/share/drivers/netapp/dataontap/client/client_cmode.py

```diff
@@ -3273,8 +3273,19 @@ def unmount_volume(self, volume_name, force=False, wait_seconds=30):
         raise exception.NetAppException(msg % msg_args)
 
     @na_utils.trace
-    def delete_volume(self, volume_name):
+    def delete_volume(self, volume_name, force_delete):
         """Deletes a volume."""
+        if force_delete:
+            res = self.send_request(
+                'system-cli', [{'priv': 'advanced'}, {'args': [
+                    {'arg': 'volume'}, {'arg': 'destroy'},
+                    {'arg': '-vserver'}, {'arg': self.vserver},
+                    {'arg': '-volume'}, {'arg': volume_name},
+                    {'arg': '-force'}, {'arg': 'True'}
+                    ]}], enable_tunneling=False)
+            if res.get_child_content('cli-result-value') == '1':
+                return
+        # Fallback in-case force-delete fails or delete without force_delete
         self.send_request('volume-destroy', {'name': volume_name})
```

Comment:
Please also log as warning when force deletion fails.
Won't there be an error raise before this point if force deletion fails? Should it be catched?

Reply:
No error throwing since we execute command and we receive reply (in case of failure, reason of failure as well). So it might have catched in wrapper layer